p2p/src/utils/requester/requester.rs

```diff
@@ -143,12 +131,6 @@ impl<E: Clock + GClock + Rng + Metrics, P: PublicKey> Requester<E, P> {
                 continue;
             }
 
-            // Check if rate limit is exceeded (and update rate limiter if not)
-            if let Err(limit) = self.rate_limiter.check_key(participant) {
```

Comment:
I think this change would mean the first request would always get routed to the "best" peer (no shuffling). This means we'd get bottlenecked by their rate limit on all new requests.

Reply:
Updated to retain the old requester logic and just remove the explicit rate limits in the reshare example. That rate limit in the requester now just serves as a way to spill over requests into lower-priority peers if a bunch of requests are sent at once if the new `rate_limit_outbound` flag is set, but felt better to retain the old behavior rather than exposing the rate limited status of the peers' outbound channel in discovery/lookup.